### PR TITLE
buildscript: fold header/path matching tests to all

### PR DESCRIPTION
**Pull request by menghanl**
Depends on https://github.com/grpc/grpc/pull/25892

### CODE REVIEW

**Comment by menghanl** [comment]
Do not submit until grpc/grpc#25892 is merged

**Comment by menghanl** [comment]
This is ready to merge

**Review by voidzcy — APPROVED** [review]

**Pull request merged**
